packages/legend-query/src/stores/QueryParameterState.ts

```diff
@@ -0,0 +1,262 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import format from 'date-fns/format';
+import addDays from 'date-fns/addDays';
+import type { InstanceValue, Type } from '@finos/legend-graph';
+import {
+  EnumValueExplicitReference,
+  EnumValueInstanceValue,
+  Enumeration,
+  CollectionInstanceValue,
+  PrimitiveType,
+  PrimitiveInstanceValue,
+  GenericType,
+  GenericTypeExplicitReference,
+  Multiplicity,
+  PRIMITIVE_TYPE,
+  VariableExpression,
+} from '@finos/legend-graph';
+import {
+  addUniqueEntry,
+  deleteEntry,
+  Randomizer,
+  uuid,
+} from '@finos/legend-shared';
+import { observable, makeObservable, action } from 'mobx';
+import type { QueryBuilderState } from './QueryBuilderState';
+import { DATE_FORMAT, DATE_TIME_FORMAT } from '@finos/legend-application';
+
+export enum QUERY_BUILDER_PARAMETER_TREE_DND_TYPE {
+  VARIABLE = 'VARIABLE',
+}
+
+export interface QueryBuilderParameterDragSource {
+  variable: ParameterState;
+}
+
+export const createMockEnumerationProperty = (
+  enumeration: Enumeration,
+): string =>
+  new Randomizer().getRandomItemInCollection(enumeration.values)?.name ?? '';
+
+const createMockPrimitiveProperty = (
+  primitiveType: PrimitiveType,
+  propertyName: string,
+): string | number | boolean => {
+  const randomizer = new Randomizer();
+  switch (primitiveType.name) {
+    case PRIMITIVE_TYPE.BOOLEAN:
+      return randomizer.getRandomItemInCollection([true, false]) ?? true;
+    case PRIMITIVE_TYPE.FLOAT:
+      return randomizer.getRandomFloat();
+    case PRIMITIVE_TYPE.DECIMAL:
+      return randomizer.getRandomDouble();
+    case PRIMITIVE_TYPE.NUMBER:
+    case PRIMITIVE_TYPE.INTEGER:
+      return randomizer.getRandomWholeNumber(100);
+    // NOTE that `Date` is the umbrella type that comprises `StrictDate` and `DateTime`, but for simplicity, we will generate `Date` as `StrictDate`
+    case PRIMITIVE_TYPE.DATE:
+    case PRIMITIVE_TYPE.STRICTDATE:
+      return format(
+        randomizer.getRandomDate(
+          new Date(Date.now()),
+          addDays(Date.now(), 100),
+        ),
+        DATE_FORMAT,
+      );
+    case PRIMITIVE_TYPE.DATETIME:
+      return format(
+        randomizer.getRandomDate(
+          new Date(Date.now()),
+          addDays(Date.now(), 100),
+        ),
+        DATE_TIME_FORMAT,
+      );
+    case PRIMITIVE_TYPE.STRING:
+    default:
+      return `${propertyName} ${randomizer.getRandomWholeNumber(100)}`;
+  }
+};
+
+export class ParameterState {
```

Comment:
Why don't we do `QueryParameterState`?

Reply:
we have a QueryParameterState. This represents the state for each parameter.

Reply:
so why don't we do `QueryParametersState` and `QueryParameterState`?